src/components/ExternalLink/index.tsx

```diff
@@ -15,10 +15,27 @@ const ExternalLink: React.FC<ExternalLinkProps> = ({
   href,
   className
 }) => {
+  const [label, setLabel] = React.useState('');
+
+  useEffect(() => {
```

Comment:
This pattern of iterating through all the links looking for the current one feels a little clunky.  Have you tried using a ref to accomplish this instead?  Something along the lines of 
```
const ExternalLink: React.FC<ExternalLinkProps> = ({
  children,
  href,
  className
}) => {
  const [label, setLabel] = React.useState('');
  const linkRef = useRef<HTMLAnchorElement>(null);

  useEffect(() => {
    if (linkRef.current) {
      setLabel(linkRef.current.innerText);
    }
  }, []);

  return (
    <a
      href={href}
      className={className}
      aria-label={label + ' (opens in new tab)'}
      rel="noopener noreferrer"
      target="_blank"
      onClick={() => {
        trackLink(href);
      }}
      ref={linkRef}
    >
      {children}
    </a>
  );
};
```

Reply:
That's significantly less clunky, thank you.